ko: add page (#22150)
tldr-pages/tldr: pages/common/ko.md

# ko

> Build and deploy Go applications as container images on Kubernetes.
> More information: <https://ko.build/reference/ko/>.

> - Build and publish a container image from an import path:

Reviewer: ```suggestion
- Build and publish a container image from a Go package import path:
```

> `ko build {{./cmd/app}}`

Reviewer: ```suggestion
`ko build {{import_path}}`
```

- Apply Kubernetes manifests with Go image references resolved to digests:

> `ko apply -f {{path/to/manifest.yaml}}`

Reviewer 2: ```suggestion
`ko apply {{[-f|--filename]}} {{path/to/manifest.yaml}}`
```

- Create Kubernetes resources with resolved image references:

> `ko create -f {{path/to/manifest.yaml}}`

Reviewer: ```suggestion
`ko create {{[-f|--filename]}} {{path/to/manifest.yaml}}`
```

- Print resolved Kubernetes manifests without applying them:

> `ko resolve -f {{path/to/manifest.yaml}}`

Reviewer: ```suggestion
`ko resolve {{[-f|--filename]}} {{path/to/manifest.yaml}}`
```

> - Run a Go binary directly on Kubernetes:

Reviewer: ```suggestion
- Build and run a Go package on Kubernetes:
```

> `ko run {{./cmd/app}}`

Reviewer: ```suggestion
`ko run {{import_path}}`
```

- Delete Kubernetes resources defined in a manifest:

> `ko delete -f {{path/to/manifest.yaml}}`

Reviewer: ```suggestion
`ko delete {{[-f|--filename]}} {{path/to/manifest.yaml}}`
```

- Log in to a container registry:

> `ko login {{registry.example.com}} -u {{username}} -p {{password}}`

Reviewer 2: ```suggestion
`ko login {{registry.example.com}} {{[-u|--username]}} {{username}} {{[-p|--password]}} {{password}}`
```
